### PR TITLE
Add patterns to OIDC identity mapping

### LINKED ISSUE #145: Importing existing `platform_oidc_identity_mapping` results in invalid resource configuration
**Describe the bug**
I'm attempting to import an existing identity mapping to inspect the resulting syntax in the Terraform config. The imported resource in the state does not match the actual mapping in the GUI and also has values not allowed by the provider.

Identity mapping in the GUI:
Image

Resource in state after import:
```
{
      "mode": "managed",
      "type": "platform_oidc_identity_mapping",
      "name": "setup",
      "provider": "provider[\"registry.terraform.io/jfrog/platform\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "claims_json": "{\"roles\":[\"jfrog_user\"]}",
            "description": "OAuth used for client setup",
            "name": "setup",
            "priority": 1,
            "provider_name": "jfrog-setup",
            "token_spec": {
              "audience": "jfrt@* jfac@* jfmc@* jfmd@* jfevt@* jfxfer@* jflnk@* jfint@* jfwks@*",
              "expires_in": 60,
              "scope": "",
              "username": null
            }
          },
          "sensitive_attributes": []
        }
      ]
    }
```
**Requirements for and issue**
Provider version: 1.15.1
Terraform version: 1.9.3
Artifactory version: 7.98.2

**Expected behavior**
Resource imported with correct values.

### PR DESCRIPTION
**Pull request by alexhung**
Closes #145 

* Add 'username_pattern' and 'groups_pattern' attributes
* Add validation to 'username' attribute to ensure 'scope' attribute is also 
* Don't set 'scope' attribute unless it's not empty string
* Update documentation to include the same

### CODE REVIEW

File: docs/resources/oidc_identity_mapping.md [hunk 1]
```diff
@@ -70,15 +126,14 @@ resource "platform_oidc_identity_mapping" "my-github-oidc-group-identity-mapping
...
 
-Required:
-
-- `scope` (String) Scope of the token. Must start with `applied-permissions/user`, `applied-permissions/admin`, `applied-permissions/roles:`, or `applied-permissions/groups:`. Group names must be comma-separated, double quotes wrapped, e.g. `applied-permissions/groups:\"readers\",\"my-group\",` Role permissions must be comma-separated, double quotes wrapped, e.g. `applied-permissions:roles:"Developer","Viewer". `username` is also required when setting role permission.
-
 Optional:
 
 - `audience` (String) Sets of (space separated) the JFrog services to which the mapping applies. Default value is `*@*`, which applies to all services.
 - `expires_in` (Number) Token expiry time in seconds. Default value is 60.
+- `groups_pattern` (String) Provide a pattern which is used to map OIDC groups to Artifactory groups.
```
Comment (danielmkn) [inline docs/resources/oidc_identity_mapping.md:133]: Should you add an example here? It should be inside of `{{}}`

File: docs/resources/oidc_identity_mapping.md [hunk 2]
```diff
@@ -70,15 +126,14 @@ resource "platform_oidc_identity_mapping" "my-github-oidc-group-identity-mapping
...
+- `scope` (String) Scope of the token. Must start with `applied-permissions/user`, `applied-permissions/admin`, `applied-permissions/roles:`, or `applied-permissions/groups:`. Group names must be comma-separated, double quotes wrapped, e.g. `applied-permissions/groups:\"readers\",\"my-group\",` Role permissions are only applicable when in project scope and must be comma-separated, double quotes wrapped, e.g. `applied-permissions:roles:<project-key>:"Developer","Viewer". `username` is also required when setting role permission.
 - `username` (String) User name of the OIDC user. Not applicable when `scope` is set to `applied-permissions/groups`. Must be set when `scope` is set to `applied-permissions/roles`.
+- `username_pattern` (String) Provide a pattern which is used to map OIDC user to Artifactory user.
```
Comment (danielmkn) [inline docs/resources/oidc_identity_mapping.md:136]: Should you add an example here? It should be inside of {{}}

**Review by danielmkn — APPROVED** [review]

File: docs/resources/oidc_identity_mapping.md (hunk 1 above)
Comment (alexhung) [reply docs/resources/oidc_identity_mapping.md:133]: I added example in the section above.

File: docs/resources/oidc_identity_mapping.md (hunk 2 above)
Comment (alexhung) [reply docs/resources/oidc_identity_mapping.md:136]: Same, look at the TF example

**Pull request merged**